src/misc/fs_utils_posix.cpp

```diff
@@ -162,11 +162,16 @@ constexpr int PermissionsRW = S_IWUSR | S_IWGRP | S_IWOTH | PermissionsRO;
 
 // Attributes 'hidden', 'system', and 'archive' are always taken from the
 // host extended attributes.
-constexpr uint8_t XattrReadMask = 0b0010'0110; // hidden, system, archive
-// Attributes 'read-only' and 'directory' are stored in extended attributes,
-// but not used by DOSBox, for these we are always using host file system
-// attribute bits.
-constexpr uint8_t XattrWriteMask = 0b0001'0001 | XattrReadMask;
+// XattrReadMask = bitflags for read-only, hidden, system, and archive
+constexpr uint8_t XattrReadMask = 0b0010'0111;
+// Attributes 'read-only' and 'directory' are stored in extended attributes.
+// For files, read-only is checked using host file system permissions.
+// For directories, read-only is checked by either file system permission or the extended attribute.
+// We do not set file system permissions when a DOS program marks a directory as read-only.
+// MS-DOS does not prevent new files from being created inside read-only directories.
+// Linux does prevent this so we must not set the host directory read-only.
+// XattrWriteMask = bitflag for directory OR'd with XattrReadMask
+constexpr uint8_t XattrWriteMask = 0b0001'0000 | XattrReadMask;
```

Comment:
Nice explanatory comment 🎖️ One more nitpicking: there should be blank lines between the constants, but I'm approving it now 😄

Reply:
Added